ehrapy/api/io/read.py

```diff
@@ -162,15 +158,64 @@ def _read(
                 raw_anndata, columns_obs_only = DataReader.read_csv(
                     filename, delimiter, index_column, columns_obs_only, cache  # type: ignore
                 )
+                # cache results if wanted
+                if cache:
+                    if not path_cache.parent.is_dir():
+                        path_cache.parent.mkdir(parents=True)
+                    return DataReader._write_cache(raw_anndata, path_cache, columns_obs_only)  # type: ignore
+
+            # read from pdf
```

Comment:
Theoretically a useless comment since the next line already states that this section is about PDF reading :)